knife/spec/unit/knife/core/ui_spec.rb

```diff
@@ -430,6 +430,15 @@ def self.from_hash(hsh)
           expect(@ui.format_for_display(input)).to eq({ nil => { "keys;with spaces;open;doors;with many.dots" => "when asked" } })
         end
       end
+
+      it "should return field separator in attribute_field_separator" do
```

Comment:
I think I'd rather see that `create_result_filter_from_attributes` responds to a set/unset `attribute_field_separator` vs. testing the set/get of the field separator itself.